patches/minecraft/net/minecraft/world/item/Item.java.patch

```diff
@@ -42,6 +42,21 @@
     public final int m_41462_() {
        return this.f_41371_;
     }
+@@ -158,11 +_,12 @@
+    }
+ 
+    public int m_142158_(ItemStack p_150900_) {
+-      return Math.round(13.0F - (float)p_150900_.m_41773_() * 13.0F / (float)this.f_41371_);
++      return Math.round(13.0F - (float)p_150900_.m_41773_() * 13.0F / (float)this.getMaxDamage(p_150900_));
+    }
+ 
+    public int m_142159_(ItemStack p_150901_) {
+-      float f = Math.max(0.0F, ((float)this.f_41371_ - (float)p_150901_.m_41773_()) / (float)this.f_41371_);
++      int stackMaxDamage = this.getMaxDamage(p_150901_);
++      float f = Math.max(0.0F, ((float)stackMaxDamage - (float)p_150901_.m_41773_()) / (float)stackMaxDamage);
```

Comment:
Wouldn't it make sense to just convert the int to a float once and supply that value to the local variable?

Reply:
I did think about that, but decided to go ahead with the current way unless told otherwise. Shall I also remove the redundant `(float)` casts as well?

Reply:
I don't see why not. The argument against is that the logic doesn't encompass any changes, but it shouldn't necessarily matter as the patch can always be replaced between versions if it happens to get updated.